lib/core/request.js

```diff
@@ -307,21 +307,11 @@ function processHeader (request, key, val) {
     key.toLowerCase() === 'transfer-encoding'
   ) {
     throw new InvalidArgumentError('invalid transfer-encoding header')
-  } else if (
-    key.length === 10 &&
-    key.toLowerCase() === 'connection'
-  ) {
-    throw new InvalidArgumentError('invalid connection header')
   } else if (
     key.length === 10 &&
     key.toLowerCase() === 'keep-alive'
   ) {
     throw new InvalidArgumentError('invalid keep-alive header')
-  } else if (
-    key.length === 7 &&
-    key.toLowerCase() === 'upgrade'
-  ) {
-    throw new InvalidArgumentError('invalid upgrade header')
   } else if (
```

Comment:
Not sure about these... why are they required?

Reply:
```
   5.   The request MUST contain an |Upgrade| header field whose value
        MUST include the "websocket" keyword.

   6.   The request MUST contain a |Connection| header field whose value
        MUST include the "Upgrade" token.
```

Reply:
The client adds them automatically through upgrade option

Reply:
Reverting these changes.